Library/Homebrew/os/mac/hardware.rb

```diff
@@ -93,12 +97,15 @@ def arch_64_bit
   # Returns an array that's been extended with ArchitectureListExtension,
   # which provides helpers like #as_arch_flags and #as_cmake_arch_flags.
   def universal_archs
-    # Building 64-bit is a no-go on Tiger, and pretty hit or miss on Leopard.
-    # Don't even try unless Tigerbrew's experimental 64-bit Leopard support is enabled.
-    if MacOS.version <= :leopard && !MacOS.prefer_64_bit?
-      [arch_32_bit].extend ArchitectureListExtension
+    # Support for running on ppc was dropped after Leopard
+    if MacOS.version > :leopard
+      [:x86_64, :i386].extend ArchitectureListExtension
+    # Building 64-bit is pretty hit or miss on Tiger and Leopard.
+    # Don't even try unless Tigerbrew's experimental 64-bit support is enabled.
+    elsif ENV["HOMEBREW_PREFER_64_BIT"]
```

Comment:
That's a good point. I'll add a new helper to encapsulate this, something like `MacOS.allow_64_bit`